### DISCUSSION TITLE
A way to implement custom filtering?

### DISCUSSION THREAD

**Discussion by vikdotdev** [piotrmurach/tty-prompt#182] [Q&A]
Let's say I have the following class:
```
class Document
  attr_reader :description, :updated_at

  def initialize(description, updated_at)
    @description = description
    @updated_at = updated_at
  end

  def to_s
    "#{description} Last updated at #{timestamp}"
  end
end
```

And here's the list I feed the prompt with:
```
list = [Document.new(...), Document.new(...)]
# some description1 Last updated at [somedate].
# some description2 Last updated at [somedate].

prompt = TTY::Prompt.new
prompt.select('', list, filter: true)
```

I want to filter based on the description only, and ignore that "Last updated at ..." bit. In other words - "Last updated at ..." should only serve as a label, and not filterable text.

Can it be achieved with existing functionality? Is my approach wrong?

Thanks

**Comment by piotrmurach** [comment]
Hi Viktor,

The short answer is no, this cannot be achieved with how the `select` works. As to answer your question about your approach. I'm not sure whether the "Last updated at.." is important to be included in the label. The most immediate solution would be to skip including that part of the label. Hope this helps.
